tasks/main.yml

```diff
@@ -63,14 +63,59 @@
     state: "{{ item.state | default(omit) }}"
     __report_changed: "{{ not __firewall_previous_replaced }}"
   loop: "{{ firewall is mapping | ternary([firewall], firewall) |
-    map('dict2items') | map('difference', __previous) | select |
+    map('dict2items') | map('difference', __previous) |
+    map('difference', __detailed) | select |
     map('items2dict') | list }}"
   vars:
     __previous:
       - key: previous
         value: replaced
+    __detailed:
+      - key: detailed
+        value: true
+      - key: detailed
+        value: false
   register: firewall_lib_result
 
+- name: gather firewall ansible fact with args
+  vars:
+    detailed: detailed
+    fw: "{{ firewall is mapping | ternary([firewall], firewall) }}"
+  block:
+    - name: gather firewall config information
+      vars:
+        __previous:
+          - key: previous
+            value: replaced
+      firewall_lib_facts:
+        detailed: "{{ item.detailed | default(False) }}"
+      loop: "{{ fw | map('dict2items') | map('difference', __previous) |
+        select | map('items2dict') | list }}"
+      register: __firewalld_facts
+
+    - name: update firewalld_config fact
+      vars:
+        firewalld_facts: "{{ __firewalld_facts.results[0] }}"
+      set_fact:
+        firewall_config: "{{ firewalld_facts.firewall_config }}"
+  when: >-
+    firewall != None and firewall | length == 1 and
+    detailed in fw[0].keys()
```

Comment:
if `fw[0]` is a `dict` then you should be able to use `detailed in fw[0]`

Reply:
The ansible recommendation for multiple `and` clauses is to use the multi-line form of `when`:
```yaml
  when:
    - not firewall is none
    - firewall | length == 1
    - detailed in fw[0]
```